python/setup.py

```diff
@@ -9,6 +9,9 @@
 except FileNotFoundError:
     long_description = ""
 
+with open(os.path.join(os.path.dirname(__file__), "requirements.txt"), encoding="utf-8") as f:
```

Comment:
```suggestion
with open(os.path.join(os.path.dirname(__file__), "requirements.in"), encoding="utf-8") as f:
```

Reply:
I'm not sure this'd be the intended source for requirements? I mean, `requirements.in` contains the unpinned requirement versions; `requirements.txt` is the compiled version with all transitive dependencies pinned to exact versions.

Reply:
Yes, the standard of Python packaging is to have loose versions in `setup.py` and pinned in `requirements.txt`. They pretend to work the same way as `package.json` and  `package-lock.json` in Node or `go.mod` and `go.sum` in Go.

Reply:
Hmm – I haven't bumped into that standard.

If this package is installed with `pip install labours`, it would then end up using non-pinned versions. When would `requirements.txt` be used then?

Reply:
requirements.txt is for the development environment / docker. https://packaging.python.org/discussions/install-requires-vs-requirements/